From: Wang Nan <wangnan0@huawei.com>
Subject: memory-hotplug: ia64: suitable memory should go to ZONE_MOVABLE

References: fate#316483
Patch-mainline: No, expected 3.17

This patch introduces zone_for_memory() to arch_add_memory() on ia64 to
ensure new, higher memory added into ZONE_MOVABLE if movable zone has
already setup.

Signed-off-by: Wang Nan <wangnan0@huawei.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Mel Gorman <mgorman@suse.de>
---

 arch/ia64/mm/init.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff -puN arch/ia64/mm/init.c~memory-hotplug-ia64-suitable-memory-should-go-to-zone_movable arch/ia64/mm/init.c
--- a/arch/ia64/mm/init.c~memory-hotplug-ia64-suitable-memory-should-go-to-zone_movable
+++ a/arch/ia64/mm/init.c
@@ -631,7 +631,8 @@ int arch_add_memory(int nid, u64 start,
 
 	pgdat = NODE_DATA(nid);
 
-	zone = pgdat->node_zones + ZONE_NORMAL;
+	zone = pgdat->node_zones +
+		zone_for_memory(nid, start, size, ZONE_NORMAL);
 	ret = __add_pages(nid, zone, start_pfn, nr_pages);
 
 	if (ret)
